src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRValueProvider.cs

```diff
@@ -9,23 +9,26 @@ namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
 {
     internal class SignalRValueProvider : IValueProvider
     {
-        private object value;
+        private readonly object value;
 
-        public SignalRValueProvider(object value)
+        private SignalRValueProvider(object value, Type valueType)
         {
             this.value = value;
-        }
 
-        public Task<object> GetValueAsync()
-        {
-            return Task.FromResult(value);
+            Type = valueType;
         }
 
-        public string ToInvokeString()
-        {
-            return value?.ToString();
-        }
+        internal static IValueProvider Create<T>(T value) where T : class
+            => value == null
+                ? (IValueProvider) SignalRNullValueProvider<T>.Instance
+                : new SignalRValueProvider(value, typeof(T));
 
         public Type Type { get; }
+
+        public Task<object> GetValueAsync()
+            => Task.FromResult(value);
```

Comment:
@wanlwanl is this a "hot" path? Would it be better to create a field holding a `Task.FromResult(value)` and set in in the constructor?

Reply:
Did the change I meant. If no good, I can revert.